### ISSUE TITLE
The cosine_decay_learning_rate scheduler does not decay after warmup

### ISSUE THREAD

**Issue by haimat** [tensorflow/models#9776]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [x] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md

## 2. Describe the bug

I am using the TF2 research object detection API with an EfficientDet D3 model for my training. The optimizer is defined in my `pipeline.config` file like this:

    optimizer {
      adam_optimizer {
        learning_rate {
          cosine_decay_learning_rate {
            learning_rate_base: 0.08
            total_steps: 300000
            warmup_learning_rate: 0.001
            warmup_steps: 250
          }
        }
      }
      use_moving_average: false
    }

So I would assume the learning rate would go up to 0.08 until step 250, afterwards it would slowly go down again until end of training at step 30,000 - is that assumption correct?

However, the learning rate chart in Tensorboard looks like this:

[![Tensorboard Learning Rate Chart][1]][1]

So the learning rate is sticking to 0.08 after step 250 has been reached.
I tried to let that run for hours - the learing rate won't go down at all.

What I am missing here?

  [1]: https://i.stack.imgur.com/wcJ5Y.png

## 3. Steps to reproduce

I Just download the E3 model from model zoo and used the config linked below on my own dataset.

## 4. Expected behavior

I would assume that the learning rate goes down after step 250.

## 5. Additional context

[The whole `pipeline.config` file can be found here](https://gist.github.com/haimat/9a373f7b021bbfd6c69c3e8134fce27c).

## 6. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04 LTS
- TensorFlow installed from (source or binary): binary installed via PIP
- TensorFlow version (use command below): v2.4.0-49-g85c8b2a817f 2.4.1
- Python version: 3.8.5
- CUDA/cuDNN version: 11.2
- GPU model and memory: Tesla P100-PCIE

**Comment by haimat** [comment]
Problem solved, solution can be found on [Stack Overflow](https://stackoverflow.com/questions/66472522/problems-with-the-tf2-learning-rate-scheduler).
